### PR TITLE
Milestone 3: Creating a data model

### PR DESCRIPTION
**Pull request by medaminedev66**
Build the project schema:
- Create and run the necessary migration files.
- Table and column names match the ERD diagram.
- Foreign keys are included.
- All columns that are foreign keys have a corresponding index.

### CODE REVIEW

File: db/migrate/20211203194905_add_user_ref_to_like.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToLike < ActiveRecord::Migration[6.1]
+  def change
+    add_reference :likes, :user, null: false, foreign_key: true
```
Comment (henatan99) [inline db/migrate/20211203194905_add_user_ref_to_like.rb:3]: - [ ] Again, the reference name should be `author`. Please refer the above similar comment for the other migration. :pray:

File: db/migrate/20211203194815_add_user_ref_to_post.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPost < ActiveRecord::Migration[6.1]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
```
Comment (henatan99) [inline db/migrate/20211203194815_add_user_ref_to_post.rb:3]: - [ ] Again, the reference name should be `author`. Please refer the above similar comment for the other migration. :pray:

File: db/migrate/20211203194425_add_user_ref_to_comment.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToComment < ActiveRecord::Migration[6.1]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
```
Comment (henatan99) [inline db/migrate/20211203194425_add_user_ref_to_comment.rb:3]: - [ ] The reference is not directly `user`, it is instead `author` to table `user`. Kindly see the below example on how to make such migration for adding reference with a different name than the reference model class name and make changes in your database. :pray: 

image

File: db/schema.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    t.index ["user_id"], name: "index_comments_on_user_id"
+  end
+
+  create_table "likes", force: :cascade do |t|
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.bigint "user_id", null: false
+    t.bigint "post_id", null: false
+    t.index ["post_id"], name: "index_likes_on_post_id"
+    t.index ["user_id"], name: "index_likes_on_user_id"
```
Comment (henatan99) [inline db/schema.rb]: - [ ] After applying the requested changes in the migrations, this should look like: 

image

File: db/schema.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  enable_extension "plpgsql"
+
+  create_table "comments", force: :cascade do |t|
+    t.text "Text"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.bigint "user_id", null: false
+    t.bigint "post_id", null: false
+    t.index ["post_id"], name: "index_comments_on_post_id"
+    t.index ["user_id"], name: "index_comments_on_user_id"
```
Comment (henatan99) [inline db/schema.rb]: - [ ] After applying the requested changes in the migrations, this should look like: 

image

File: db/schema.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+  create_table "posts", force: :cascade do |t|
+    t.string "Title"
+    t.text "Text"
+    t.integer "CommentsCounter"
+    t.integer "LikesCounter"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.bigint "user_id", null: false
+    t.index ["user_id"], name: "index_posts_on_user_id"
```
Comment (henatan99) [inline db/schema.rb]: - [ ] After applying the requested changes in the migrations, this should look like: 

image

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @medaminedev66 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by medaminedev66** [comment]
Hi @henatan99. Thanks for reviewing my pull request. Ok, I will implement the required changes.

**Comment by Davidosky007** [comment]
## Hi @medaminedev66 👋, 

## Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

thumbs-up-gif


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**